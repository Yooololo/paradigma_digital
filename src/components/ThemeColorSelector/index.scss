.ThemeColorSelector {
  .themeColor {
    display: flex;
    align-items: center;
    -webkit-tap-highlight-color: transparent;
  }
  .theme__fill,
  .theme__icon {
    pointer-events: none;
    transition: transform $animation ease-in-out;
  }
  .theme__fill {
    background-color: var(--bg);
    display: block;
    mix-blend-mode: difference;
    position: fixed;
    inset: 0;
    height: 100%;
    transform: translateX(-100%);
  }
  .theme__icon,
  .theme__toggle {
    z-index: 1;
  }
  .theme__icon,
  .theme__icon__part {
    position: absolute;
  }
  .theme__icon {
    display: block;
    top: 2.125em;
    left: 0.225em;
    width: 1.5em;
    height: 1.5em;
  }
  .theme__icon__part {
    border-radius: 50%;
    box-shadow: 0.4em -0.4em 0 0.5em $gold inset;
    top: calc(50% - 0.75em);
    left: calc(50% - 0.75em);
    width: 1em;
    height: 1em;
    transition: box-shadow $animation ease-in-out,
      opacity $animation ease-in-out, transform $animation ease-in-out;
    transform: scale(0.5);
  }
  .theme__icon__part ~ .theme__icon__part {
    background-color: $gold;
    border-radius: 0.05em;
    box-shadow: none;
    top: calc(50% - 0.25em);
    left: calc(50% - 0.3em);
    transform: rotate(0) translateY(0.5em);
    transform-origin: 50% 0;
    width: 0.1em;
    height: 0.2em;
  }
  .theme__icon__part:nth-child(3) {
    transform: rotate(45deg) translateY(0.5em);
  }
  .theme__icon__part:nth-child(4) {
    transform: rotate(90deg) translateY(0.5em);
  }
  .theme__icon__part:nth-child(5) {
    transform: rotate(135deg) translateY(0.5em);
  }
  .theme__icon__part:nth-child(6) {
    transform: rotate(180deg) translateY(0.5em);
  }
  .theme__icon__part:nth-child(7) {
    transform: rotate(225deg) translateY(0.5em);
  }
  .theme__icon__part:nth-child(8) {
    transform: rotate(270deg) translateY(0.5em);
  }
  .theme__icon__part:nth-child(9) {
    transform: rotate(315deg) translateY(0.5em);
  }
  .theme__label,
  .theme__toggle,
  .theme__toggle__wrap {
    position: relative;
  }
  .theme__toggle,
  .theme__toggle::before {
    display: block;
  }
  .theme__toggle {
    &.light {
      background-color: $blue;
    }

    border-radius: 25% / 50%;
    box-shadow: 0 0 0 0.125em var(--primaryT);
    padding: 0.25em;
    width: 6em;
    height: 3em;
    appearance: none;
    -webkit-appearance: none;
    cursor: pointer;
    transition: background-color $animation ease-in-out,
      box-shadow 0.15s ease-in-out, transform $animation ease-in-out;
  }

  .light.theme__toggle::before {
    background-color: $grey;
  }
  .theme__toggle::before {
    border-radius: 50%;
    content: "";
    width: 2.5em;
    height: 2.5em;
    transition: background-color $animation ease-in-out,
      transform $animation ease-in-out;
  }
  .theme__toggle:focus {
    box-shadow: 0 0 0 0.125em var(--primary);
    outline: transparent;
  }
  /* Checked */
  .dark.theme__toggle:checked {
    background-color: $black;
  }
  .theme__toggle:checked::before,
  .theme__toggle:checked ~ .theme__icon {
    transform: translateX(3em);
  }
  .dark.theme__toggle:checked::before {
    background-color: $grey;
  }
  .theme__toggle:checked ~ .theme__fill {
    transform: translateX(0);
  }
  .theme__toggle:checked ~ .theme__icon .dark.theme__icon__part:nth-child(1) {
    transform: scale(1) translateY(-0.05em) translateX(0) rotate(10deg);
    box-shadow: 0.15em -0.15em 0 0.15em $white inset;
  }
  .theme__toggle:checked
    ~ .theme__icon
    .theme__icon__part
    ~ .theme__icon__part {
    opacity: 0;
  }
  .theme__toggle:checked ~ .theme__icon .theme__icon__part:nth-child(2) {
    transform: rotate(45deg) translateY(0.8em);
  }
  .theme__toggle:checked ~ .theme__icon .theme__icon__part:nth-child(3) {
    transform: rotate(90deg) translateY(0.8em);
  }
  .theme__toggle:checked ~ .theme__icon .theme__icon__part:nth-child(4) {
    transform: rotate(135deg) translateY(0.8em);
  }
  .theme__toggle:checked ~ .theme__icon .theme__icon__part:nth-child(5) {
    transform: rotate(180deg) translateY(0.8em);
  }
  .theme__toggle:checked ~ .theme__icon .theme__icon__part:nth-child(6) {
    transform: rotate(225deg) translateY(0.8em);
  }
  .theme__toggle:checked ~ .theme__icon .theme__icon__part:nth-child(7) {
    transform: rotate(270deg) translateY(0.8em);
  }
  .theme__toggle:checked ~ .theme__icon .theme__icon__part:nth-child(8) {
    transform: rotate(315deg) translateY(0.8em);
  }
  .theme__toggle:checked ~ .theme__icon .theme__icon__part:nth-child(9) {
    transform: rotate(360deg) translateY(0.8em);
  }
  .theme__toggle__wrap {
    margin: 0 0.75em;
  }
}
