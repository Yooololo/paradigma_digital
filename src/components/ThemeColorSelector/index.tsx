import { useContext } from "react";
import { ThemeColorContext } from "../../context/ThemeColorWrapper";

export default function ThemeColorSelector() {
  const { themeColor, setThyThemeColor } = useContext(ThemeColorContext);

  return (
    <div className="ThemeColorSelector">
      <label htmlFor="themeColor" className={themeColor}>
        <span className="theme__toggle__wrap">
          <input
            type="checkbox"
            className={`theme__toggle ${themeColor}`}
            id="themeColor"
            role="switch"
            name="themeColor"
            value={themeColor}
            onChange={() => setThyThemeColor(themeColor)}
            checked={themeColor === "dark"}
          />
          <span className={`theme__fill ${themeColor}`} />
          <span className={`theme__icon ${themeColor}`}>
            <span className={`theme__icon__part ${themeColor}`}></span>
            <span className={`theme__icon__part ${themeColor}`}></span>
            <span className={`theme__icon__part ${themeColor}`}></span>
            <span className={`theme__icon__part ${themeColor}`}></span>
            <span className={`theme__icon__part ${themeColor}`}></span>
            <span className={`theme__icon__part ${themeColor}`}></span>
            <span className={`theme__icon__part ${themeColor}`}></span>
            <span className={`theme__icon__part ${themeColor}`}></span>
            <span className={`theme__icon__part ${themeColor}`}></span>
          </span>
        </span>
      </label>
    </div>
  );
}
