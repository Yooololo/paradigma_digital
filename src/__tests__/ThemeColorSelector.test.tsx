import { cleanup, render } from "@testing-library/react";
import { IntlProvider } from "react-intl";
import English from "../languages/en.json";
import ThemeColorSelector from "../components/ThemeColorSelector";

afterEach(cleanup);

test("LanguageSelector renders correctly", () => {
  const { container } = render(
    <IntlProvider locale="en-US" messages={English}>
      <ThemeColorSelector />
    </IntlProvider>
  );
  expect(container).toMatchSnapshot();
  expect(container.querySelector(".ThemeColorSelector")).toBeTruthy();
  expect(container.querySelectorAll(".theme__icon__part")).toHaveLength(9);
});
