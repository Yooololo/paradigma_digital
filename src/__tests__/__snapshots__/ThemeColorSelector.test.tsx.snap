// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LanguageSelector renders correctly 1`] = `
<div>
  <div
    class="ThemeColorSelector"
  >
    <label
      class="dark"
      for="themeColor"
    >
      <span
        class="theme__toggle__wrap"
      >
        <input
          checked=""
          class="theme__toggle dark"
          id="themeColor"
          name="themeColor"
          role="switch"
          type="checkbox"
          value="dark"
        />
        <span
          class="theme__fill dark"
        />
        <span
          class="theme__icon dark"
        >
          <span
            class="theme__icon__part dark"
          />
          <span
            class="theme__icon__part dark"
          />
          <span
            class="theme__icon__part dark"
          />
          <span
            class="theme__icon__part dark"
          />
          <span
            class="theme__icon__part dark"
          />
          <span
            class="theme__icon__part dark"
          />
          <span
            class="theme__icon__part dark"
          />
          <span
            class="theme__icon__part dark"
          />
          <span
            class="theme__icon__part dark"
          />
        </span>
      </span>
    </label>
  </div>
</div>
`;
